packages/client-config/API.md

```diff
@@ -45,8 +87,76 @@ export type AuthClientConfig = {
     };
 };
 
-// @public
-export type ClientConfig = Partial<AnalyticsClientConfig & AuthClientConfig & GeoClientConfig & GraphqlClientConfig & NotificationsClientConfig & StorageClientConfig & PlatformClientConfig & CustomClientConfig>;
+// @public (undocumented)
+enum AuthorizationType {
+    // (undocumented)
+    AmazonCognitoUserPools = "AMAZON_COGNITO_USER_POOLS",
+    // (undocumented)
+    APIKey = "API_KEY",
+    // (undocumented)
+    AwsIam = "AWS_IAM",
+    // (undocumented)
+    AwsLambda = "AWS_LAMBDA",
+    // (undocumented)
+    OpenidConnect = "OPENID_CONNECT"
+}
+
+// @public (undocumented)
+enum AuthorizationType_2 {
+    // (undocumented)
+    AmazonCognitoUserPools = "AMAZON_COGNITO_USER_POOLS",
+    // (undocumented)
+    APIKey = "API_KEY",
+    // (undocumented)
+    AwsIam = "AWS_IAM",
+    // (undocumented)
+    AwsLambda = "AWS_LAMBDA",
+    // (undocumented)
+    OpenidConnect = "OPENID_CONNECT"
+}
+
+// @public (undocumented)
+enum AwsRegion {

```

Comment:
I would suggest that we type aws region properties as string. Providing an enum type becomes a never-ending game of catchup

Reply:
added a comment in the schema doc for PMs to look at.

Reply:
that's a good callout to pivot this to a string. It appears CDK went a similar path and opted not to surface an enum